lms/migrations/versions/48955a83f040_events_backfill.py

```diff
@@ -0,0 +1,74 @@
+"""
+Backfills the events table using the data from  `lti_launches`.
+
+Also renames the old table and create a view with the same name to keep
+any reports based on it backward compatible.
+
+
+Revision ID: 48955a83f040
+Revises: 42b39684836f
+Create Date: 2022-08-01 14:30:34.211522
+
+"""
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "48955a83f040"
+down_revision = "42b39684836f"
+
+
+def upgrade():
+    conn = op.get_bind()
+
+    launch_event_type_id = conn.execute(
+        "SELECT id from event_type WHERE type = 'configured_launch'"
+    ).fetchone()[0]
+
+    conn.execute(
+        f"""
+        INSERT INTO event
+            (
+                timestamp,
+                type_id,
+                application_instance_id,
+                course_id
+            )
+        SELECT
+                lti_launches.created,
+                {launch_event_type_id},
+                application_instances.id,
+                grouping.id
+        FROM lti_launches
+        JOIN application_instances
+            ON lti_launches.lti_key = application_instances.consumer_key
+        JOIN grouping
+            ON context_id = grouping.lms_id and grouping.application_instance_id = application_instances.id and grouping.type = 'course'
+    """
```

Comment:
I think there's a benefit to querying performance if the physical layout of the rows matches the logical one. So it's ideal if the rows are inserted in timestamp order. (Something about that here https://www.cybertec-postgresql.com/en/column-correlation-explained/).

So we might want to ensure we sort the rows by `created` date here before we insert them.

Also we'd ideally migrate all the old rows and start capturing the new ones at exactly the same moment after, but I'm not sure how we'd do that.

Reply:
I'm wondering if using the launch event type is going to complicate our lives here, as it means we'll have the same type, but two very different types of rows when it comes to pre-and-post data collection.

I'm not sure how this will look, but I'm thinking of trying to write SQL and then having to remember that most of the rows will have NULLs for various fields. I'm not sure if that will cause more or less problems.

Maybe we can save ourselves a headache by either using a special "legacy_login" type or not migrating at all? I don't know which is best.